rasa/shared/core/training_data/story_writer/yaml_story_writer.py

```diff
@@ -206,9 +206,10 @@ def process_user_utterance(
             result[KEY_USER_INTENT] = user_utterance.intent_name
 
         if hasattr(user_utterance, "inline_comment"):
-            result.yaml_add_eol_comment(
-                user_utterance.inline_comment(), KEY_USER_INTENT
-            )
+            if user_utterance.predicted_intent != user_utterance.intent["name"]:
```

Comment:
why do we need this check now?

Reply:
Because as mentioned [here](https://github.com/RasaHQ/rasa/pull/8376#discussion_r633310981) for the entities, this line
`if hasattr(user_utterance, "inline_comment"):`
will always be `True` even if the wrong prediction is in entities or intent. So it will not distinguish. It will always return `True` because in any wrong prediction we will end up with an object of class `WronglyClassifiedUserUtterance` which always `has attribute ` the `inline_comment`. 

So with this check, if the wrong prediction is in entities but the intent is correct, I do not print any comment in the `intent` line.